test/abstractarray.jl

```diff
@@ -548,12 +548,19 @@ function test_primitives(::Type{T}, shape, ::Type{TestAbstractArray}) where T
     @test firstindex(B, 1) == firstindex(A, 1) == first(axes(B, 1))
     @test firstindex(B, 2) == firstindex(A, 2) == first(axes(B, 2))
 
-    # isassigned(a::AbstractArray, i::Int...)
+    @test !isassigned(B)
+    # isassigned(a::AbstractArray, i::Integer...)
     j = rand(1:length(B))
     @test isassigned(B, j)
     if T == T24Linear
         @test !isassigned(B, length(B) + 1)
     end
+    # isassigned(a::AbstractArray, i::CartesianIndex)
+    @test isassigned(B, first(CartesianIndices(B)))
+    ind = last(CartesianIndices(B))
+    @test !isassigned(B, ind + oneunit(ind))
+    # isassigned(a::AbstractArray, i::Union{Integer,CartesianIndex}...)
+    @test isassigned(B, Int16.(first.(axes(B)))..., CartesianIndex(1,1))
 
```

Comment:
Let's add a few more tests:

```suggestion
    @test !isassigned(B, Bool.(first.(axes(B)))..., CartesianIndex(1,1))
    @test !isassigned(B, Bool.(first.(axes(B)))...)
    @test !isassigned(B, true)
    @test !isassigned(B, false)
```

Reply:
Oh, actually these throw at present with `Bool` arguments, so perhaps we should test for that.

Reply:
I was worried about that — hence the tests. :)

Reply:
I mean throwing for `Bool` indices is the current behavior on master. This PR appears to change this, which should probably be fixed. However, we should perhaps update the tests to `@test_throws`?

Reply:
Makes perfect sense.  All I know is that I had a tingling spidey sense here.  This looks great.